RevenueCatUI/Helpers/Package+VariableDataProvider.swift

```diff
@@ -24,7 +52,71 @@ extension Package: VariableDataProvider {
     }
 
     var localizedPrice: String {
-        return self.storeProduct.localizedPriceString
+        if shouldRoundPrices {
+            return self.localizedPriceRounded
+        } else {
+            return self.storeProduct.localizedPriceString
+        }
+    }
+
+    var localizedPriceRounded: String {
+        if self.storeProduct.priceFormatter {
+            roundPriceWithFormatter()
+        } else {
+            roundPriceWithSearchAndReplace()
+        }
+    }
+
+    func roundPriceWithFormatter() -> String {
+        guard let formatter = self.storeProduct.priceFormatter else {
+            return self.storeProduct.localizedPriceString
+        }
+
+        guard let priceToRound = formatter.number(from: self.storeProduct.localizedPriceString) else {
+            return self.storeProduct.localizedPriceString
+        }
+
+        let originalMaximumFractionalDigits = formatter.maximumFractionDigits
+
+        defer { formatter.maximumFractionDigits = originalMaximumFractionalDigits }
+        formatter.maximumFractionDigits = 0
```

Comment:
Can we create a new formatter instead of modifying the existing one? Modifying state scares me a little bit 😅

Reply:
Good call, yes!